include/seqan3/io/sequence/sequence_file_out.hpp

```diff
@@ -0,0 +1,900 @@
+ * writing to another, because you don't have to configure the output file to match the input file, it will just work:
+ *
+ * ```cpp
+ * sequence_file_in   fin{"input.fasta", fields<field::ID, field::SEQ_QUAL>{}};
+ * sequence_file_out fout{"output.fasta"}; // doesn't have to match the configuration
+ *
+ * for (auto & r : fin)
+ * {
+ *     if // r fulfills some criterium
+ *     {
+ *         fout.push_back(r);
+ *     }
+ * }
+ * ```
+ *
+ * ### Writing record-wise in batches
+ *
+ * You can write multiple records at once, by assigning to the file:
+ *
+ * ```cpp
+ * sequence_file_out fout{"/tmp/my.fasta"};
+ *
+ * std::vector<std::tuple<dna5_vector, std::string>> range
+ * {
+ *     { "ACGT"_dna5, "First" },
+ *     { "NATA"_dna5, "2nd" },
+ *     { "GATA"_dna5, "Third" }
+ * }; // a range of "records"
+ *
+ * fout = range; // will iterate over the records and write them
+ * ```
+ * ### File I/O pipelines
+ *
+ * Record-wise writing in batches also works for writing from input files directly to output files, because input
+ * files are also input ranges in SeqAn:
+ *
+ * ```cpp
+ * // file format conversion in one line:
+ * sequence_file_out fout{"output.fasta"} = sequence_file_in{"input.fastq"};
+ *
+ * // or in pipe notation:
+ * sequence_file_in{"input.fastq"} | sequence_file_out{"output.fasta"};
+ * ```
+ *
+ * This can be combined with file-based views to create I/O pipelines:
+ *
+ * ```cpp
+ * sequence_file_in{"input.fastq"} | view::minimum_average_quality_filter(20)
+ *                                 | view::minimum_sequence_length_filter(50)
+ *                                 | ranges::view::take(5)
+ *                                 | sequence_file_out{"output.fasta"};
+ * ```
+ *
+ * ### Column-based writing
+ *
+ * The record-based interface treats the file as a range of tuples (the records), but in certain situations
+ * you might have the data as columns, i.e. a tuple-of-ranges, instead of range-of-tuples.
+ *
+ * You can use column-based writing in that case, it uses operator=() :
+ *
+ * ```cpp
+ *
+ * struct data_storage_t
+ * {
+ *     concatenated_sequences<dna5_vector>  sequences;
+ *     concatenated_sequences<std::string>  ids;
+ * };
+ *
+ * data_storage_t data_storage; // a global or globally used variable in your program
+ *
+ * // ... in your file writing function:
+ *
+ * sequence_file_out fout{"/tmp/my.fasta"};
+ *
+ * fout = std::tie(data_storage.sequences, data_storage.ids);
+ * ```
+ *
+ * ### Formats
+ *
+ * TODO give overview of formats, once they are all implemented
+ */
+
+template <detail::fields_concept selected_field_ids_ = fields<field::SEQ, field::ID, field::QUAL>,
+          detail::type_list_of_sequence_file_out_formats_concept valid_formats_ =
+              type_list<sequence_file_format_fasta /*, ...*/>,
+          ostream_concept<char> stream_type_ = std::ofstream>
+class sequence_file_out
+{
+public:
+    /*!\name Template arguments
```

Comment:
Doxygen groups all of these under the level-1-headline "Public Types" which is the doxygen word for member types. I used the same names here as for sequence_file_in

Reply:
But template arguments???
Like I associate it with documentation of the template arguments of the class. But this is basically a forwarding declaration of the template parameters. Well, what's about `Interface Types`... 🤕

Reply:
Hm, well, we define a whole lot of member types, and I think "Template arguments" with the description "Exposed as member types for public access." is about as verbose as it gets? 

"Interface types" otoh could be really confusing since we speak of the "row-based interface" and the "column based interface" and the template arguments are the only group that actually belongs to neither of them :thinking: